src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/vehicle_info_pending_resumption_test.cc

```diff
@@ -0,0 +1,609 @@
+  EXPECT_EQ(ext->PendingSubscriptions().GetData().size(), 0u);
+}
+
+TEST_F(VehicleInfoPendingResumptionHandlerTest,
+       OneAppSeveralVehicleDataResponseOveralResultFailed) {
+  auto mock_app = CreateApp(1);
+  auto ext = CreateExtension(*mock_app);
+  ext->AddPendingSubscription("gps");
+  ext->AddPendingSubscription("speed");
+
+  uint32_t corr_id = 345;
+  auto request = CreateVDRequest(corr_id);
+  ON_CALL(mock_message_helper_,
+          CreateModuleInfoSO(VehicleInfo_SubscribeVehicleData, _))
+      .WillByDefault(Return(request));
+
+  EXPECT_CALL(subscribe_catcher_, subscribe(_, _));
+  // TODO save cid and fid of the subscription
+  EXPECT_CALL(event_dispatcher_mock_, raise_event(_));
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_, _)).Times(1);
+
+  auto response =
+      CreateVDError(corr_id, hmi_apis::Common_Result::ABORTED, "error message");
+  // TODO use error message type instead of result code
+  application_manager::event_engine::Event event(
+      VehicleInfo_SubscribeVehicleData);
+  event.set_smart_object(response);
+  resumption_handler_->HandleResumptionSubscriptionRequest(
+      *ext, get_subscriber(), *mock_app);
+
+  // TODO check that raized the same fid and cid as subscribed
+  resumption_handler_->on_event(event);
+  EXPECT_FALSE(ext->isSubscribedToVehicleInfo("gps"));
+  EXPECT_FALSE(ext->isSubscribedToVehicleInfo("speed"));
+  EXPECT_EQ(ext->PendingSubscriptions().GetData().size(), 0u);
+}
+
+TEST_F(VehicleInfoPendingResumptionHandlerTest, TwoAppsOneSharedDataSuccess) {
+  auto mock_app = CreateApp(1);
+  auto mock_app2 = CreateApp(2);
+  auto ext = CreateExtension(*mock_app);
+  auto ext2 = CreateExtension(*mock_app2);
+  ext->AddPendingSubscription("gps");
+  ext2->AddPendingSubscription("gps");
+
+  uint32_t corr_id = 345;
+  auto request = CreateVDRequest(corr_id);
+  ON_CALL(mock_message_helper_,
+          CreateModuleInfoSO(VehicleInfo_SubscribeVehicleData, _))
+      .WillByDefault(Return(request));
+
+  EXPECT_CALL(subscribe_catcher_, subscribe(_, _)).Times(2);
+  // TODO save cid and fid of the subscription
+  EXPECT_CALL(event_dispatcher_mock_, raise_event(_)).Times(2);
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_, _)).Times(1);
+  const std::map<std::string, hmi_apis::Common_VehicleDataResultCode::eType>
+      subscriptions_result = {
+          {"gps", hmi_apis::Common_VehicleDataResultCode::VDRC_SUCCESS},
+      };
+
+  auto response = CreateVDResponse(
+      hmi_apis::Common_Result::SUCCESS, subscriptions_result, corr_id);
+  // TODO use error message type instead of result code
+  application_manager::event_engine::Event event(
+      VehicleInfo_SubscribeVehicleData);
+  event.set_smart_object(response);
+  resumption_handler_->HandleResumptionSubscriptionRequest(
+      *ext, get_subscriber(), *mock_app);
+  resumption_handler_->HandleResumptionSubscriptionRequest(
+      *ext2, get_subscriber(), *mock_app2);
+  // TODO check that raized the same fid and cid as subscribed
```

Comment:
When are you planning to process all of this ToDo's? As I understand, now they are out of scope of this review?

Reply:
I hope to take care about it on monday. But I would not like to stuck with it and delay other activities. Also this activity may be aplitted with other developers